lib/views/detailed_todo_page.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+
+import 'package:todo_app/models/todo.dart';
+
+class DetaledTodoPage extends StatefulWidget {
+  const DetaledTodoPage({Key? key}) : super(key: key);
+
+  @override
+  State<DetaledTodoPage> createState() => _DetaledTodoPageState();
+}
+
+class _DetaledTodoPageState extends State<DetaledTodoPage> {
+  @override
+  Widget build(BuildContext context) {
+    final todo = context.extractArguments() as Todo;
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Details'),
+        leading: Row(
+          children: [
+            GestureDetector(
+              child: const Icon(
+                Icons.navigate_before,
+              ),
+              onTap: () => Navigator.pop(context, todo),
+            ),
```

Comment:
take a look on [this](https://api.flutter.dev/flutter/material/IconButton-class.html) i think it would be useful in this case